tools/ducktape/run_ducktape.py

```diff
@@ -0,0 +1,501 @@
+                capture_output=True,
+                text=True,
+            )
+            if result.stdout.strip():
+                logs_dir = self.src_dir / "logs"
+                logs_dir.mkdir(exist_ok=True)
+                with open(logs_dir / "azurite.log", "w") as f:
+                    subprocess.run(
+                        ["docker", "logs", "azurite"],
+                        stdout=f,
+                        stderr=subprocess.STDOUT,
+                        check=False,
+                    )
+        except Exception as e:
+            self.log(f"Could not fetch azurite logs: {e}")
+
+        # Stop docker compose
+        self.run_cmd(
+            ["docker", "compose", "down"], cwd=compose_dir, env=env, check=False
+        )
+
+    def run(self) -> int:
+        """Run the full ducktape test pipeline."""
+        try:
+            # Build and link artifacts
+            if not self.args.skip_build:
+                self.build_ducktape_package()
+                self.link_bazel_artifacts()
+
+            # Build docker image
+            self.build_test_docker_image()
+
+            # Start cluster and run tests
+            self.start_compose_cluster()
+            self.generate_cluster_config()
+
+            exit_code = self.run_ducktape_tests()
+
+            return exit_code
+
+        finally:
+            if not self.args.no_cleanup:
+                self.stop_compose_cluster()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run Redpanda ducktape integration tests",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog=__doc__,
+    )
+
+    # Build settings
+    parser.add_argument(
+        "--build-type",
+        default="fastbuild",
+        choices=["release", "debug", "fastbuild"],
+        help="Build type (default: fastbuild)",
+    )
+    parser.add_argument("--bazel-args", default="", help="Additional bazel arguments")
+    parser.add_argument(
+        "--skip-build",
+        action="store_true",
+        help="Skip building bazel package (use existing artifacts)",
+    )
+
+    # Docker settings
+    parser.add_argument(
+        "--skip-build-docker-image",
+        action="store_true",
+        help="Skip building docker image",
+    )
+    parser.add_argument(
+        "--node-count", type=int, help="Number of test nodes (default: CPU count)"
+    )
+
+    # Test settings
+    parser.add_argument(
+        "--max-parallel",
+        type=int,
+        default=1,
+        help="Maximum parallel tests (default: 1)",
+    )
+    parser.add_argument(
+        "--repeat",
+        type=int,
+        default=1,
+        help="Number of times to repeat tests (default: 1)",
+    )
+    parser.add_argument(
+        "--test-timeout",
+        type=int,
+        default=1800000,
+        help="Test timeout in milliseconds (default: 1800000)",
+    )
+    parser.add_argument(
+        "--no-cleanup",
```

Comment:
make this the default? takes too long to stop otherwise and gets in the edit-build-test dopamine loop

maybe in the future we do a daemon like bazel and stop on idle or something but for now prefer manual stop

i always have the vtools compose running; restart it once a quarter for some hacking only

Reply:
I split this into a few subcommands. PTAL